compiler/lib/php_from_rehp.re

```diff
@@ -980,48 +1139,122 @@ and statement = (curOut, input, x) => {
       let (eOut, eMapped) = expression(input, e);
       (outAppend(curOut, eOut), Throw_statement(eMapped));
     | Rehp.Switch_statement(e, l, def, l') =>
-      let (eOut, eMapped) = expression(input, e);
-      let (dOut, dMapped) = optOutput(statements(curOut, input), def);
+      let nextInput = {vars: input.vars, label: Switch};
```

Comment:
I thought that `label` was modeling label scopes, but it was unexpected to see a `Switch` here since it doesn't actually impact scope.

Reply:
Sorry about the confusion, the names for this variable and its variants are bad. 

It represents whether the current AST is immediately enclosed within a labelled loop, a regular loop, or a switch. By "immediate" I mean this:

```
function f() {
  x = 3; // "input.label" is NoLabel when processing this statement
  a: for () {
    x = 3; // "input.label" is LabelledLoop("a")
    for () {
      x = 3; // "input.label" is UnlabelledLoop
    }
    switch () {
      case 0:
        x = 3; // "input.label" is Switch
      case 1:
        for () {
          x = 3; // "input.label" is UnlabelledLoop
        }
    }
  }
}
```

It's necessary to know whether a continue_statement is enclosed within a switch in order to convert `continue` to `break; continue_label="#";`.

It's useful to know whether something is inside of a labelled loop, so that something like this:

```
a: for () {
  for () {
    continue a;
  } 
  x = 3;
}
```

Can be compiled to this:

```
for () {
  continue_label = null;
  for () {
    continue_label = "a";
    break;
  } 
  if (continue_label === "a") {
    continue;
  }
  x = 3;
}
```

Instead of this:

```
for () {
  continue_label = null;
  for () {
    continue_label = "a";
    break;
  } 
  if (continue_label === "a") {
    continue;
  } else if (continue_label != null) {
    break;
  }
  x = 3;
}
```

If we know that the label of all "unshielded" continue statements inside loop X is the same as the label of a loop Y that encloses loop X, then it's not necessary to add a check to break after loop X.